src/hivpy/circumcision.py

```diff
@@ -0,0 +1,182 @@
+
+    def init_birth_circumcision_born(self, population, date):
+        """
+        Initialise circumcision at birth for all born males.
+
+        This is an alternative birth circumcision initialisation
+        method to `init_birth_circumcision_all` and requires
+        the use of `update_birth_circumcision` at every time step
+        to work as expected.
+        """
+        male_born_population = population.index[(population[col.SEX] == SexType.Male)
+                                                & (population[col.AGE] >= 0.25)]
+        r = rng.uniform(size=len(male_born_population))
+        circumcision = r < self.prob_birth_circ
+        population.loc[male_born_population, col.CIRCUMCISED] = circumcision
+        # all circumcised males get a circumcision date of the start of the simulation
+        population.loc[population[col.CIRCUMCISED], col.CIRCUMCISION_DATE] = date
+
+    def update_birth_circumcision(self, population, time_step, date):
+        """
+        Update birth circumcision for newly born males.
+        COVID disruption is factored in.
+
+        Requires birth circumcision to be initialised with
+        `init_birth_circumcision_born` instead of
+        `init_birth_circumcision_all` in order to work as expected.
+        """
+        # covid disruption causes circumcision probability to be 0
+        if (self.covid_disrup_affected != 1) & (self.vmmc_disrup_covid != 1):
+            # assumes ages have already been incremented
+            newborn_males = population.index[(population[col.SEX] == SexType.Male)
+                                             & (population[col.AGE] >= 0.25)
+                                             & (population[col.AGE] - time_step.days / 365 < 0.25)]
+            r = rng.uniform(size=len(newborn_males))
+            circumcision = r < self.prob_birth_circ
+            population.loc[newborn_males, col.CIRCUMCISED] = circumcision
+            # newly circumcised males get the current date set as their circumcision date
+            circ_newborn_males = population.index[population[col.CIRCUMCISED]
+                                                  & population[col.CIRCUMCISION_DATE].isnull()]
+            population.loc[circ_newborn_males, col.CIRCUMCISION_DATE] = date
+
+    def update_vmmc(self, pop):
+        """
+        Update voluntary medical male circumcision intervention.
+        COVID disruption is factored in.
+        """
+        # keep date in circumcision for now for the sake of
+        # the function passed to transform_group
+        self.date = pop.date  # should year be float instead of int?
+
+        # no further circumcision
+        if (self.vmmc_disrup_covid == 1) \
+           | ((self.year_interv <= self.date.year)
+              & (self.circ_inc_rate_year_i == 2)) \
+           | ((self.year_interv + 5 <= self.date.year)
+              & (self.circ_inc_rate_year_i == 4)):
+            self.test_link_circ_prob = 0
```

Comment:
```suggestion
        if (self.vmmc_disrup_covid == 1) \
           | ((self.year_interv <= self.date.year) & (self.circ_inc_rate_year_i == 2)) \
           | ((self.year_interv + 5 <= self.date.year) & (self.circ_inc_rate_year_i == 4)):
            self.test_link_circ_prob = 0
```
I personally find this formatting a bit easier to read as it requires less parsing of brackets between lines, but up to you which you prefer

Reply:
I agree that this formatting is easier to read, but flake8 will complain about the line being too long (> 100 characters).

Reply:
I think we should probably consider making the allowable line length a little longer (maybe 120 characters) given that a) everyone's IDE can display more than 100 characters these days and b) we have a lot of long variables names which makes keeping things under the 100 character limit difficult. This is a common problem and had led to a lot of weird formatting in some of the modules